azurerm/modules/azurerm-data/adls-de/adls.tf

```diff
@@ -0,0 +1,28 @@
+resource "azurerm_storage_account" "adls_default" {
+  name                     = var.storage_account_name
+  resource_group_name      = var.resource_group_name
+  location                 = var.resource_group_location
+  account_kind             = var.account_kind
+  account_tier             = var.account_tier
+  account_replication_type = var.adls_account_replication_type
+  is_hns_enabled           = var.hns_enabled
+}
+
+resource "azurerm_storage_data_lake_gen2_filesystem" "adls_lake_default" {
+  for_each = var.adls_containers
+
+  name               = each.key
+  storage_account_id = azurerm_storage_account.adls_default.id
+}
+
+resource "azurerm_storage_container" "adls_container" {
```

Comment:
If this module is just for creating the data lake storage (rather than a general storage container), then this isn't needed. Just for a recap in the data platform we will need 2 storage accounts ultimately creating:
 - One for the data lake - HNS enabled, and `azurerm_storage_data_lake_gen2_filesystem`
 - One for the config data - without HNS, and `azurerm_storage_container`
 
So, we could make the module generic so it can support either type.

Reply:
@adurkan-amido 

I have amended the code now. 

One for the data lake - HNS enabled, and azurerm_storage_data_lake_gen2_filesystem
One for the config data - without HNS, and azurerm_storage_container and a blob inside the container for the config data. 

The one for config data now is fully optional, and can be managed via a set of variables: 

```
  create_additional_storage   = true
  create_additional_container = true
  create_additional_blob      = true 
```